radiography/src/main/java/radiography/ViewTreeRenderingVisitor.kt

```diff
@@ -0,0 +1,43 @@
+package radiography
+
+import android.annotation.TargetApi
+import android.os.Build.VERSION_CODES
+import android.view.View
+import android.view.ViewGroup
+
+/**
+ * A [TreeRenderingVisitor] that renders [View]s and their children which match [viewFilter] using
+ * [stateRenderers].
+ */
+internal class ViewTreeRenderingVisitor(
+  private val stateRenderers: List<StateRenderer<*>>,
+  private val viewFilter: ViewFilter
+) : TreeRenderingVisitor<View>() {
+
+  override fun RenderingScope.visitNode(node: View) {
+    description.viewToString(node)
+
+    if (node !is ViewGroup) return
+
+    // Capture this value, since it might change while we're iterating.
+    val childCount = node.childCount
+    for (index in 0 until childCount) {
+      // Child may be null, if children were removed by another thread after we captured the child
+      // count. getChildAt returns null for invalid indices, it doesn't throw.
```

Comment:
Can be simplified once https://github.com/square/radiography/pull/30 is merged (single thread)